drivers/sdhc/sdhc_renesas_ra.c

```diff
@@ -0,0 +1,746 @@
+				(read_sector_size / SDHI_MAX_BLOCK_SIZE);
+
+			if (SDMMC_CARD_TYPE_MMC == priv->sdmmc_ctrl.device.card_type) {
+				/*
+				 * If c_size is 0xFFF, then sector_count should be obtained from the
+				 * extended CSD. Set it to 0 to indicate it should come from the
+				 * extended CSD later.
+				 */
+				if (SDHI_PRV_SECTOR_COUNT_IN_EXT_CSD == p_csd_reg.csd_v1_b.c_size) {
+					priv->sdmmc_ctrl.device.sector_count = 0U;
+				}
+			}
+		}
+
+#if SDHI_CFG_SD_SUPPORT_ENABLE
+		else if (SDHI_PRV_CSD_VERSION_2_0 == csd_version) {
+			priv->sdmmc_ctrl.device.sector_count =
+				(p_csd_reg.csd_v2_b.c_size + 1U) * SDHI_PRV_BYTES_PER_KILOBYTE;
+		} else {
+			/* Do Nothing */
+		}
+
+		if (SDHI_PRV_CSD_VERSION_1_0 == csd_version) {
+			/* Get the minimum erasable unit (in 512 byte sectors). */
+			priv->sdmmc_ctrl.device.erase_sector_count =
+				p_csd_reg.csd_v1_b.sector_size + 1U;
+		} else
+#endif
+		{
+			/*
+			 * For SDHC and SDXC cards, there are no erase group restrictions.
+			 * Using the eMMC TRIM operation, there are no erase group restrictions.
+			 */
+			priv->sdmmc_ctrl.device.erase_sector_count = 1U;
+		}
+		break;
+	case SD_APP_SEND_OP_COND:
+		ra_cmd.opcode |= SDHI_PRV_CMD_C_ACMD;
+		ret = sdhc_ra_send_cmd(priv, &ra_cmd, retries);
+		if (ret < 0) {
+			goto end;
+		}
+		sdmmc_response_t response;
+		/* get response of ACMD41 (R3) */
+		response.status = priv->sdmmc_ctrl.p_reg->SD_RSP10;
+		/*  Initialization complete? */
+		if (response.r3.power_up_status) {
+			/* High capacity card ? */
+			/*  0 = SDSC, 1 = SDHC or SDXC */
+			priv->sdmmc_ctrl.sector_addressing =
+				(response.r3.card_capacity_status > 0U);
+			priv->sdmmc_ctrl.device.card_type = SDMMC_CARD_TYPE_SD;
+		}
+		priv->sdmmc_ctrl.initialized = true;
+		break;
+	case SD_SWITCH:
+		/* Check app cmd */
+		if (priv->app_cmd && cmd->opcode == SD_APP_SET_BUS_WIDTH) {
+			/* ACMD41*/
+			ra_cmd.opcode |= SDHI_PRV_CMD_C_ACMD;
+			ret = sdhc_ra_send_cmd(priv, &ra_cmd, retries);
+			if (ret < 0) {
+				goto end;
+			}
+		} else {
+			/* SD SWITCH CMD6*/
+			fsp_err = r_sdhi_read_and_block(&priv->sdmmc_ctrl, ra_cmd.opcode,
+							ra_cmd.arg, data->block_size);
+			ret = err_fsp2zep(fsp_err);
+			if (ret < 0) {
+				goto end;
+			}
+			memcpy(ra_cmd.data, priv->sdmmc_ctrl.aligned_buff, 8);
+			priv->sdmmc_event.transfer_completed = false;
+			break;
+		}
+		break;
+
+	/* Read write with data */
+	case SD_APP_SEND_SCR:
+		ra_cmd.opcode = cmd->opcode | SDHI_PRV_CMD_C_ACMD;
+		fsp_err = r_sdhi_read_and_block(&priv->sdmmc_ctrl, ra_cmd.opcode, ra_cmd.arg,
+						data->block_size);
+
+		if (fsp_err != 0) {
+			ret = -ETIMEDOUT;
+			goto end;
+		}
+		memcpy(ra_cmd.data, priv->sdmmc_ctrl.aligned_buff, 8);
+		priv->sdmmc_event.transfer_completed = false;
+		break;
+	case SD_READ_SINGLE_BLOCK:
+	case SD_READ_MULTIPLE_BLOCK:
+		/* Configure the transfer interface for reading.*/
+		fsp_err = r_sdhi_transfer_read(&priv->sdmmc_ctrl, ra_cmd.sector_count,
```

Comment:
Ok, I *think* I understand what is going on here (please correct me if I'm wrong):
- the `sector_count` is being used as the size of block for the DTC (which seems like a DMA controller) to transfer
- the `block_size` is being used as the number of blocks for the DTC to transfer

Is this all right? If so, we should not be setting the `block_size` as a devicetree property. We should be using the `block_size` field from the `struct sdhc_data` provided for this request. Same thing for the other locations where we use a hardcoded `block-size`.

Reply:
@danieldegrasse : just a little confirmation, those two have opposite meanings
- The `sector_count` is being used as the number of blocks for the DTC to transfer
- The `block_size `  is being used as the size of blocks for the DTC to transfer.

I see, I have removed the `block_size `setting as a devicetree property and used the `block_size` from struct `sdhc_data` instead. Thanks very much for letting me know.

Reply:
Sure, thanks for the clarification, sorry I missed that while looking at the DTC implementation